### PR TITLE
First draft of HUD for vector mask tool

### PR DESCRIPTION
**Pull request by pineapplespatula**
Adding a hud for the vector mask super select. needs some more work with seth before its ready to merge though

### CODE REVIEW

File: src/js/actions/mask.js [hunk 1]
```diff
@@ -139,6 +144,80 @@ define(function (require, exports) {
...
+
+        if (currentLayers.isEmpty()) {
+            return Promise.resolve();
+        }
+
+        var currentLayer = currentLayers.first(),
+            bounds = currentLayer.bounds;
+            
+        return descriptor.batchPlayObjects([vectorMaskLib.deleteVectorMask(),
+            vectorMaskLib.makeCircularBoundsWorkPath(bounds),
```
Comment (pineapplespatula) [inline src/js/actions/mask.js]: This does not handle history properly currently. @mcilroyc what do i need to add to make it history respectful

Comment (shaoshing) [inline src/js/actions/mask.js]: Have you try adding the `historyStateInfo` option? I see multiple histories in classic after applying a rectangle or eclipse mask.

Comment (mcilroyc) [inline src/js/actions/mask.js]: what @shaoshing said.

**Comment by pineapplespatula** [comment]
@placegraphichere is working on this for a bit, I'll open it back up when its ready to be PRed

**Pull request closed without merging**